### PR TITLE
cassowary: fix source url

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
I've opened an issue upstream to ask for clarification.

File: Formula/cassowary.rb [hunk 1]
```diff
@@ -2,8 +2,9 @@ class Cassowary < Formula
   desc "Modern cross-platform HTTP load-testing tool written in Go"
   homepage "https://github.com/rogerwelin/cassowary"
   url "https://github.com/rogerwelin/cassowary/archive/v0.14.0.tar.gz"
-  sha256 "385232478b8552d56429fbe2584950bfbe42e3b611919a31075366a143aae9a9"
+  sha256 "04fbf1e563378bc37a5ad44e69a3b100647406ea2debae2d38d3c406399628fa"
```
Comment (cho-m) [inline Formula/cassowary.rb:5]: The short URL is hitting some duplicate, which is causing it to not resolve actual tarball URL. May be due to Git tag and branch with same name:
```console
❯ curl https://github.com/rogerwelin/cassowary/archive/v0.14.0.tar.gz -L
the given path has multiple possibilities: #<Git::Ref:0x00007fd87706cf50>, #<Git::Ref:0x00007fd877072428>
```

We will need to use longer URL (https://github.com/rogerwelin/cassowary/archive/refs/tags/v0.14.0.tar.gz)

File: Formula/cassowary.rb [hunk 2]
```diff
@@ -1,9 +1,10 @@
 class Cassowary < Formula
   desc "Modern cross-platform HTTP load-testing tool written in Go"
   homepage "https://github.com/rogerwelin/cassowary"
-  url "https://github.com/rogerwelin/cassowary/archive/v0.14.0.tar.gz"
+  url "https://github.com/rogerwelin/cassowary/archive/refs/tags/v0.14.0.tar.gz"
   sha256 "385232478b8552d56429fbe2584950bfbe42e3b611919a31075366a143aae9a9"
   license "MIT"
+  revision 1
```
Comment (carlocab) [inline Formula/cassowary.rb:7]: This is no longer needed, since nothing has changed for the built bottles.

**Comment by iMichka** [comment]
Fixed. Thanks for finding out about the branch.

**Review by carlocab — APPROVED** [review]

**Pull request merged**